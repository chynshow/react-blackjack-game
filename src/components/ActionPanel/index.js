import React, { useContext } from 'react';
import { AppContext } from '../../state/AppContext';

const ActionPanel = () => {
  const {
    hitAction,
    doubleDownAction,
    standAction,
    state: { stand },
  } = useContext(AppContext);
  return (
    <div className='c-action-panel'>
      <button
        className='c-btn c-action-panel__btn'
        disabled={stand}
        onClick={() => hitAction()}
      >
        Hit
      </button>
      <button
        className='c-btn c-action-panel__btn'
        onClick={() => standAction()}
      >
        {stand ? 'Next' : 'Stand'}
      </button>
      <button
        className='c-btn c-action-panel__btn'
        onClick={() => doubleDownAction()}
        disabled={stand}
      >
        Double Down
      </button>
    </div>
  );
};

export default ActionPanel;
