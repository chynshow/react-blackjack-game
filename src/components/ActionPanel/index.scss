.c-action-panel {
  &__btn {
  }
}
