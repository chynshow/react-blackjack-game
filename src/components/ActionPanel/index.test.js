import React from 'react';
import ActionPanel from './index';
import { shallow } from 'enzyme';

it('render ActionPanel component', () => {
  const c = shallow(<ActionPanel />);
  expect(c).toMatchSnapshot();
});
