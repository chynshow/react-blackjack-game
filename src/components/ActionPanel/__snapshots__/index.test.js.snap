// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ActionPanel component 1`] = `
<div
  className="c-action-panel"
>
  <button
    className="c-btn c-action-panel__btn"
    disabled={false}
    onClick={[Function]}
  >
    Hit
  </button>
  <button
    className="c-btn c-action-panel__btn"
    onClick={[Function]}
  >
    Stand
  </button>
  <button
    className="c-btn c-action-panel__btn"
    disabled={false}
    onClick={[Function]}
  >
    Double Down
  </button>
</div>
`;
